lib/src/main/java/org/asamk/signal/manager/Manager.java

```diff
@@ -189,6 +190,8 @@
     private final SignalServiceMessageReceiver messageReceiver;
     private final ClientZkProfileOperations clientZkProfileOperations;
 
+    private final ExecutorService executor = Executors.newCachedThreadPool();
+
```

Comment:
I think a single thread (`newSingleThreadExecutor()`) might be a bottleneck in scenarios where one wants to send multiple messages on a mediocre internet connection. Having read the Javadoc around Executors more carefully, I suggest using a `Executors.newCachedThreadPool()`. That way, the behavior is almost similar to before the change - with the benefit that threads are reused and that the thread pool is shutdown on closing.